tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php

```diff
@@ -1,73 +1,45 @@
 <?php
 namespace Elementor\Tests\Phpunit\Elementor\Core\Editor\Loader;
 
-use Elementor\Core\Editor\Editor;
-use Elementor\Core\Editor\Editor_V2_Experiments;
 use Elementor\Core\Editor\Loader\Editor_Loader_Factory;
 use Elementor\Core\Editor\Loader\V1\Editor_V1_Loader;
 use Elementor\Core\Editor\Loader\V2\Editor_V2_Loader;
-use Elementor\Core\Experiments\Manager as Experiments_Manager;
-use Elementor\Core\Utils\Collection;
-use Elementor\Plugin;
 use ElementorEditorTesting\Elementor_Test_Base;
 
 if ( ! defined( 'ABSPATH' ) ) {
 	exit; // Exit if accessed directly
 }
 
 class Test_Editor_Loader_Factory extends Elementor_Test_Base {
-	private $original_experiments_states;
+	public function set_up() {
+		parent::set_up();
 
-	public function setUp(): void {
-		parent::setUp();
-
-		$experiments = Plugin::$instance->experiments;
-
-		$this->original_experiments_states = Collection::make( Editor_V2_Experiments::all() )
-			->map_with_keys( fn ( $name ) => [ $name => $experiments->get_features( $name )['default'] ] )
-			->each( fn ( $state, $name ) => $experiments->set_feature_default_state( $name, Experiments_Manager::STATE_INACTIVE ) );
-	}
-
-	public function tearDown(): void {
-		parent::tearDown();
-
-		$experiments = Plugin::$instance->experiments;
-
-		$this->original_experiments_states
-			->each( fn ( $state, $name ) => $experiments->set_feature_default_state( $name, $state ) );
+		remove_all_filters( 'elementor/editor/v2/packages' );
 	}
 
-	/**
-	 * @dataProvider create_data_provider
-	 */
-	public function test_create( $experiment, $state, $expected_class ) {
+	public function test_create__returns_an_instance_of_v2_loader_when_packages_are_enqueued() {
```

Comment:
I think it should be based on the experiment and not based on the packages. The packages are being enqueued based on the experiments so just use the loader based on the same experiments.

Reply:
We should talk about it.